bftengine/src/preprocessor/PreProcessor.cpp

```diff
@@ -1457,30 +1464,34 @@ void PreProcessor::registerAndHandleClientPreProcessReqOnNonPrimary(const string
   }
 }
 
+uint32_t PreProcessor::getBufferOffset(uint16_t clientId, ReqId reqSeqNum, uint16_t reqOffsetInBatch) const {
+  return (clientId - numOfReplicas_ - numOfInternalClients_) * clientMaxBatchSize_ + reqOffsetInBatch;
```

Comment:
maybe it's worth asserting that ```(clientId - numOfReplicas_ - numOfInternalClients_)``` is not negative

Reply:
OK